src/workers/ans104-unbundler.ts

```diff
@@ -66,33 +69,46 @@ export class Ans104Unbundler {
     dataItemIndexFilterString: string;
     workerCount: number;
     maxQueueSize?: number;
+    shouldUnbundle?: () => boolean;
+    ans104Parser?: Ans104Parser;
   }) {
     this.log = log.child({ class: 'Ans104Unbundler' });
     this.filter = filter;
-    this.ans104Parser = new Ans104Parser({
-      log,
-      eventEmitter,
-      contiguousDataSource,
-      workerCount,
-      dataItemIndexFilterString,
-    });
+    this.ans104Parser =
+      ans104Parser ||
+      new Ans104Parser({
+        log,
+        eventEmitter,
+        contiguousDataSource,
+        workerCount,
+        dataItemIndexFilterString,
+      });
 
     this.workerCount = workerCount;
     this.maxQueueSize = maxQueueSize;
     this.queue = fastq.promise(
       this.unbundle.bind(this),
       Math.max(workerCount, 1),
     );
+    this.shouldUnbundle = shouldUnbundle;
   }
 
   async queueItem(
     item: UnbundleableItem,
     prioritized: boolean | undefined,
   ): Promise<void> {
+    const log = this.log.child({ method: 'queueItem', id: item.id });
+
     if (this.workerCount === 0) {
+      log.warn('Skipping data item queuing due to no workers.');
       return;
     }
-    const log = this.log.child({ method: 'queueItem', id: item.id });
+
+    if (!this.shouldUnbundle()) {
```

Comment:
Here I'm skipping it even if it's prioritized. Should we keep that way or ignore `shouldUnbundle` when the item is prioritized?

Reply:
Good question. I'd say let's ignore it if it's prioritized `shouldUnbundle`.

Reply:
[`dfe19de` (#190)](https://github.com/ar-io/ar-io-node/pull/190/commits/dfe19de7fa8e785a0ada4394bfad416c384a9151)